### ISSUE TITLE
(discussion) Phasing out the wiki for a site?

### ISSUE THREAD

**Issue by ItaloKnox** [07th-mod/guide#8]
I've been stuck with this idea for a while, but what do you guys think about phasing out the wiki to make a site? Of course, I'll do it. I'm planning something very simple (but good looking), just an extension of our wiki, a lot more organized and a space to put out announcements and some other stuff. Should also be easier to navigate overall.

So, what do you guys think?

**Comment by enumag** [comment]
Sorry I completely missed this since I was not watching the repository.

If you want to do it, I don't see any reason against it. Go right ahead!

If the website would be static we could also use github pages to setup auto deployment. There can even be a build process with travis. My blog works that way although it was admittedly a little hard to set up correctly.

**Comment by ItaloKnox** [comment]
I actually thought that was the case later. 😆 

I did some stuff about this but ended being a downgrade compared to the wiki. Had to sacrifice information to make it more readable, so for install instructions, it doesn't seem to be really good, but we can try for something else.

Also, you nailed it, that was pretty much the idea though I never used Travis despite knowing how it works.

I think the solution might be to keep both. The wiki is very useful right now, but we can use a site to show some stuff we can't do on the wiki. Not sure exactly *what*, though.

**Comment by enumag** [comment]
In that case I think we should postpone it for now. It doesn't make much sense to create a website if we don't know what to put there.

Although I'm not quite sure how it can be a downgrade from the wiki. Can't you just copy-paste it 1:1 at the beginning?

**Comment by ItaloKnox** [comment]
Well, just go back to our [install page](https://github.com/07th-mod/guide/wiki/Part-1---Voice-and-Graphics-Patch) and see how much stuff we cover there. The point was to make it a lot simpler, however, you actually gave me a nice idea. I think we can keep the wiki and just put the installer information on the site. We'll miss OSX and Linux information but these are a minority, and for Umineko I also have an installer in the works. This should work better and won't require updates very often.

**Comment by enumag** [comment]
Yeah, that page should probably be separated to multiple pages... Also I see that it's even longer now that you added an install.sh script with instructions how to use it. Great work by the way. I'm guessing that's the main logic and improving it with `wget` downloads of the zip archives should be easy in the future?

We can omit the OSX/Linux instructions on the website at first though. It's quite useless now since the patch is actually broken on OSX anyway. I'm a bit concerned if the OSX issue will be fixed. We can't really revert to the vanilla DLLs with all the we have made to the scripts since.

**Comment by ItaloKnox** [comment]
I don't really mind keeping everything on one wiki page since as long as it's in topic we can use the jumps to get where we need pretty easily, but for a website that might not be really good. Also thanks, you pretty much got the gist of it. I pushed that script really fast after seeing the OSX issue, but the plan is to extend it later to download and install any chapters. Actually, it would be nice if I could scrape the releases page to get the latest version number automatically and build the link from there, but that is something [I will look into later](http://docs.python-guide.org/en/latest/scenarios/scrape/).

I have [this theme](http://fmedias.com/straightway/index_light.html) here. There are a few features already coded in that would help a bit (accordions, tabs) and cleaning it a little seems to work for a nice little static page. We won't need most of the default stuff but I kinda like how it opens and that it looks really compact. Give me your thoughts on it. :)

And last, about the OSX issue, it really concerns me as well. I really don't want to drop support (though I don't think we necessarily will - or is the voice inserting script already outputting it ready for the new dll? Not really a big fan of it if true, that will make voices-only releases break, I guess?).

**Comment by enumag** [comment]
@ItaloKnox Instead of scraping it might be better to use [GitHub API](https://developer.github.com/v3/repos/releases/). For example here are Onikakushi releases: https://api.github.com/repos/07th-mod/onikakushi/releases

Personally even though I'm web developer I never made a site with such theme. But it looks modern and clean so I like it. As long as it's not a flash/silverlight of course. JavaScript should be enough these days. Just use some shorter times. The example you linked takes forever to fully display everything.

Voice insertion does and always will work fine with vanilla DLLs. I have no plans to change that. I want to do the new chapters the same way as before just with some additional steps at the end of the process for the recent features. Of course if you have some suggestions for changes, please tell me.

**Comment by ItaloKnox** [comment]
Yeah, I was dumb there and forgot Github has an API. I actually managed to get the link to the latest release pretty easily: ``curl -k https://api.github.com/repos/07th-mod/onikakushi/releases/latest | grep browser_download_url | cut -d '"' -f 4`` << using -k here because curl on windows is not behaving very well without the certs. Works fine on Linux without it, though.

The only issue I have is that I'm not really a god at running multiple commands like that so I'm pretty unsure how to put that link on aria2c.

About the layout, you're right. I'm using pretty low times and removed most of the stuff there. Page seems to load almost instantly so that's a win. I'll see if I can upload a copy later.

I'm glad the voice insertion is still like that, seems like there is a real base for voices-only patches. Kinda makes me want to revisit them and insert the voices that were missing. For now, I'm pretty happy with our setup and looking forward to seeing what we can do with the next chapter.
